### PR TITLE
feat(desktop): add chat archiving with settings restore

### PR DESCRIPTION
**Pull request by gannonh**
## Summary
- add session archiving support in desktop state atoms with per-workspace filtering
- show archive action on session item hover (bottom-right icon) and hide archived chats from sidebar
- add **Archived Chats** settings tab with title, archived date, project directory, and unarchive action
- add coverage for archive/unarchive behavior in renderer atom tests

## Verification
- `pnpm run typecheck`
- `npx vitest run src/renderer/atoms/__tests__/session-archive.test.ts src/renderer/components/settings/__tests__/SettingsPanel.test.tsx`
- pre-push hook suite passed (`@kata/desktop:test`)

## New Features
- **Session archiving**: Archive and restore chat sessions within your workspace
- **Git integration**: Displays current branch and pull request information in the chat footer
- **Archived chats management**: New settings panel to view and restore archived chat sessions

## Improvements
- Reorganized chat composer controls and footer layout
- Message list now limits displayed messages for performance optimization with option to view older messages
- Removed legacy permission mode UI from chat interface

<h3>Greptile Summary</h3>

This PR adds client-side chat archiving to Kata Desktop. Sessions can be archived from the sidebar hover action and restored via a new **Archived Chats** settings tab. Archive state is persisted with `atomWithStorage` (keyed by workspace path), and `visibleSessionListAtom` filters the sidebar in real time. The PR also ships several unrelated UI improvements: workspace git branch / PR URL display in the chat footer, message virtualization (`MAX_RENDERED_MESSAGES = 80`), removal of the legacy permission-mode selector, and a compact mode for `ModelSelector` and `ThinkingLevelToggle`.

- The `gh pr view` call in `readWorkspaceGitInfo` uses a 250 ms timeout — shorter than a typical GitHub API round-trip — so `pullRequestUrl` will silently return `null` on most connections and the \"Open PR\" button won't appear in practice.
- Sessions archived while `workingDirectoryAtom` is empty are stored with `projectDir = 'Unknown workspace'`, but `workspaceArchivedSessionIdsAtom` filters against `normalizeWorkspacePath('') = ''`, so those entries never match and the sessions remain visible.

<h3>Confidence Score: 5/5</h3>

Safe to merge; archive feature is correct for the normal path and all findings are P2

No P0/P1 blockers. The two logic notes (gh timeout, Unknown workspace sentinel) are edge cases that don't affect the primary archive/unarchive flow covered by the new tests. The git-info staleness is a UX quality issue.

apps/desktop/src/main/ipc.ts (gh timeout) and apps/desktop/src/renderer/atoms/session.ts (Unknown workspace sentinel)

<details><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| apps/desktop/src/renderer/atoms/session.ts | Core archive/unarchive logic: introduces archivedSessionsAtom (atomWithStorage), visibleSessionListAtom, archiveSessionAtom, and unarchiveSessionAtom — minor edge case when archiving with no workspace set |
| apps/desktop/src/main/ipc.ts | Adds workspaceGetGitInfo IPC handler; gh pr view called with 250 ms timeout which is too short for a network request |
| apps/desktop/src/renderer/components/settings/ArchivedChatsPanel.tsx | New settings tab rendering archived chats from all workspaces with unarchive action; clean and correct |
| apps/desktop/src/renderer/components/app-shell/SessionListItem.tsx | Adds archive button on hover/focus-within; wraps in memo; moves Intl formatters to module-level constants |
| apps/desktop/src/renderer/components/chat/ChatPanel.tsx | Removes PermissionModeSelector, adds git branch/PR display; git info fetched once per workspace change with stale-safe active flag |
| apps/desktop/src/renderer/components/chat/MessageList.tsx | Adds soft message virtualization (MAX_RENDERED_MESSAGES=80) with show-older button; toolsByParent moved into useMemo |
| apps/desktop/src/renderer/components/chat/MessageInput.tsx | Adds footerControls slot, auto-resize textarea via useLayoutEffect, layout polish |
| apps/desktop/src/renderer/components/ui/textarea.tsx | Converts to React.forwardRef to enable ref prop for auto-resize in MessageInput |

</details>

</details>

<h3>Flowchart</h3>

```mermaid
%%{init: {'theme': 'neutral'}}%%
flowchart TD
    A[User clicks Archive button on SessionListItem] --> B[archiveSessionAtom write]
    B --> C[Append to archivedSessionsAtom with projectDir + archivedAt]
    C --> D{Is archived session the current session?}
    D -- No --> E[Return early - sidebar updates via visibleSessionListAtom]
    D -- Yes --> F[get workspaceArchivedSessionIdsAtom now includes new entry]
    F --> G{Any remaining visible sessions?}
    G -- Yes --> H[switchSessionAtom to first visible session]
    G -- No --> I[Clear currentSessionIdAtom + reset chat state]

    J[User opens Settings - Archived Chats] --> K[archivedSessionsForSettingsAtom all workspaces sorted by archivedAt]
    K --> L[ArchivedChatsPanel renders list]
    L --> M[User clicks Unarchive]
    M --> N[unarchiveSessionAtom removes entry from archivedSessionsAtom]
    N --> O[visibleSessionListAtom updates - session reappears in sidebar]
```

<details><summary><h3>Comments Outside Diff (1)</h3></summary>

1. `apps/desktop/src/main/ipc.ts`, line 129 ([link](https://github.com/gannonh/kata/blob/c20ca51951b7b356176885fbb805a39c90687311/apps/desktop/src/main/ipc.ts#L129)) 

   <a href="#">P2</a> **`gh pr view` timeout too short for a network request**

   250 ms is far below the round-trip time of a GitHub API call on any realistic network (typically 300–2 000 ms). `runOptionalCommand` silently returns `null` on timeout, so `pullRequestUrl` will be `null` almost every time and the "Open PR" button will never appear in practice.

   Consider raising the timeout to at least 5 000 ms, or making the `gh` step a lazy/on-demand query rather than blocking the idle callback.

   

   <details><summary>Prompt To Fix With AI</summary>

   `````markdown
   This is a comment left during a code review.
   Path: apps/desktop/src/main/ipc.ts
   Line: 129

   Comment:
   **`gh pr view` timeout too short for a network request**

   250 ms is far below the round-trip time of a GitHub API call on any realistic network (typically 300–2 000 ms). `runOptionalCommand` silently returns `null` on timeout, so `pullRequestUrl` will be `null` almost every time and the "Open PR" button will never appear in practice.

   Consider raising the timeout to at least 5 000 ms, or making the `gh` step a lazy/on-demand query rather than blocking the idle callback.

   

   How can I resolve this? If you propose a fix, please make it concise.
   `````
   </details>

</details>

<details><summary>Prompt To Fix All With AI</summary>

`````markdown
This is a comment left during a code review.
Path: apps/desktop/src/main/ipc.ts
Line: 129

Comment:
**`gh pr view` timeout too short for a network request**

250 ms is far below the round-trip time of a GitHub API call on any realistic network (typically 300–2 000 ms). `runOptionalCommand` silently returns `null` on timeout, so `pullRequestUrl` will be `null` almost every time and the "Open PR" button will never appear in practice.

Consider raising the timeout to at least 5 000 ms, or making the `gh` step a lazy/on-demand query rather than blocking the idle callback.

```suggestion
    : await runOptionalCommand('gh', ['pr', 'view', '--head', branch, '--json', 'url', '--jq', '.url'], workspacePath, 5000)
```

How can I resolve this? If you propose a fix, please make it concise.

---

This is a comment left during a code review.
Path: apps/desktop/src/renderer/atoms/session.ts
Line: 378-379

Comment:
**"Unknown workspace" sentinel doesn't round-trip through `workspaceArchivedSessionIdsAtom`**

When `workingDirectoryAtom` is empty, `archiveSessionAtom` stores `projectDir = 'Unknown workspace'`. But `workspaceArchivedSessionIdsAtom` compares against `normalizeWorkspacePath('')` which is `''`, not `'Unknown workspace'`, so those archived entries never match and the sessions remain visible in the sidebar.

A simple fix is to use the same sentinel in both places:

```ts
// in workspaceArchivedSessionIdsAtom
const normalizedWorkspacePath = normalizeWorkspacePath(get(workingDirectoryAtom)) || 'Unknown workspace'
```

Or normalize both sides to the same empty-fallback before comparison.

How can I resolve this? If you propose a fix, please make it concise.

---

This is a comment left during a code review.
Path: apps/desktop/src/renderer/components/chat/ChatPanel.tsx
Line: 68-86

Comment:
**Git info fetched once; display goes stale on branch switch**

The effect's dependency array is `[workingDirectory]`, so git info is fetched once when the workspace is first loaded and never refreshed. If the user switches branches in the terminal while Kata is open, the displayed branch (and PR URL) will remain stale for the entire session.

Consider periodically re-fetching (e.g. on window focus or on a slow interval) or adding a manual refresh affordance so users can pull the current branch without switching workspaces.

How can I resolve this? If you propose a fix, please make it concise.
`````

</details>

<sub>Reviews (1): Last reviewed commit: ["feat(desktop): add chat archiving with s..."](https://github.com/gannonh/kata/commit/c20ca51951b7b356176885fbb805a39c90687311) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=28534597)</sub>

> Greptile also left **2 inline comments** on this PR.

### CODE REVIEW

File: apps/desktop/src/renderer/atoms/session.ts [hunk 1]
```diff
@@ -319,6 +372,68 @@ export const switchSessionAtom = atom(null, async (get, set, sessionId: string)
   }
 })
 
+export const archiveSessionAtom = atom(
+  null,
+  async (get, set, session: SessionListItem) => {
+    const sessionId = session.id.trim()
+    if (!sessionId) {
```
[the remark this answers is not in the document]
Comment (gannonh) [reply apps/desktop/src/renderer/atoms/session.ts:381]: Fixed in `d740fe4f`.

I aligned the workspace matching key by applying the same unknown-workspace fallback in `workspaceArchivedSessionIdsAtom` (`normalizeWorkspacePath(...) || 'Unknown workspace'`).

I also added regression coverage in `session-archive.test.ts` (`archives and hides sessions when workspace path is missing`) to ensure archived sessions are hidden correctly when no workspace path is available.

File: apps/desktop/src/renderer/atoms/session.ts [hunk 2]
```diff
@@ -319,6 +372,68 @@ export const switchSessionAtom = atom(null, async (get, set, sessionId: string)
...
+    const nextSession = get(sessionListAtom).find((entry) => !archivedSessionIds.has(entry.id))
+
+    if (nextSession) {
+      await set(switchSessionAtom, nextSession.id)
+      return
+    }
+
+    set(currentSessionIdAtom, null)
+    set(resetChatStateAtom)
+    set(resetPlanningSessionStateAtom)
```
[the remark this answers is not in the document]
Comment (gannonh) [reply apps/desktop/src/renderer/atoms/session.ts:429]: Addressed in `d740fe4f`.

`archiveSessionAtom` now explicitly blocks archiving the active session while streaming and surfaces a user-facing error (`Stop the active run before archiving this chat.`), preventing state desynchronization.

I added a regression test (`rejects archiving the active session while streaming`) to lock this behavior.
